### ISSUE TITLE
SplitCoordinator Initialization

### ISSUE THREAD

**Issue by ghost** [quickbirdstudios/XCoordinator#89]
If the window was not been appears and rootViewController of master is a UINavigationController, `SplitCoordinator.init(master: Presentable, detail: Presentable?)` works in an unexpected way.

For example:
```
import UIKit
import XCoordinator

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {
    let window: UIWindow! = UIWindow()
    let router = SomeCoordinator().anyRouter

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        router.setRoot(for: window!)
        return true
    }
}

enum SomeRoute: Route { }

class SomeCoordinator: SplitCoordinator<SomeRoute> {
    init() {
        let masterVC = UIViewController()
        masterVC.title = "Master"

        let detailVC = UIViewController()
        detailVC.title = "Detail"

        super.init(
            master: UINavigationController(rootViewController: masterVC),
            detail: detailVC
        )
    }
}
```
![Untitled](https://user-images.githubusercontent.com/42165252/57447904-d8215200-7260-11e9-8712-f7c24a7d1165.png)

I found the problem in `BaseCoordinator.performTransitionAfterWindowAppeared(_ transition: TransitionType)`. If performTransition is wrapped asynchronously, then everything works as expected.
```
private func performTransitionAfterWindowAppeared(_ transition: TransitionType) {
    guard UIApplication.shared.keyWindow == nil else {
        return performTransition(transition, with: TransitionOptions(animated: false))
    }

    var windowAppearanceObserver: Any?

    rootViewController.beginAppearanceTransition(true, animated: false)
    windowAppearanceObserver = NotificationCenter.default.addObserver(
    forName: UIWindow.didBecomeKeyNotification, object: nil, queue: .main) { [weak self] _ in
        windowAppearanceObserver.map(NotificationCenter.default.removeObserver)
        windowAppearanceObserver = nil
        DispatchQueue.main.async {
            self?.performTransition(transition, with: TransitionOptions(animated: false))
        }
        self?.rootViewController.endAppearanceTransition()
    }
}
```

**Comment by pauljohanneskraft** [comment]
Hey - thanks for finding and reporting this issue. I'm not quite sure `BaseCoordinator.performTransitionAfterWindowAppeared(_ transition: TransitionType)` is the correct place to fix this behavior as it might break similar cases for other coordinator classes... 

What you can do for now as a hotfix is performing a `.showDetail` transition in the `presented(from:)` method of your coordinator instead of specifying the detail view controller in the initializer. Please keep in mind to call `super.presented(from:)` when overriding.

**Comment by sthwlf** [comment]
Yea, I was not sure that this fix is thread safe, so I created the issue, not a pull request. A simpler hotfix is an overriding the `generateRootViewController` method for setting init view controllers.

**Comment by pauljohanneskraft** [comment]
This is addressed in XCoordinator 2.0.0, which will be released in the near future (see https://github.com/quickbirdstudios/XCoordinator/pull/122 for more information).
